lib/lte_link_control/Kconfig

```diff
@@ -365,6 +368,12 @@ config LTE_LC_TRACE
 	  what has happened and has no associated payload.
 	  It is intended to be used for debugging purposes.
 
+config LTE_EXTENDED_FEATURES
+	bool "Extended features"
+	help
+	  Enables support for extended features, notifications, and events: eDRX,
+	  Periodic TAU Pre-Warning, Neighboring Cell Measurements, Modem Sleep, Modem Domain Events.
+
```

Comment:
This is definitely a valid point that I've been considering while implementing this solution and ultimately decided to simplify to not introduce too many new Kconfig options. However, since there is interest in higher granularity I will revisit it. This means that a bigger refactor is required inside the `lte_link_control` library in a way to separate the different functionality domains. I'll mull it over.

Reply:
> However, since there is interest in higher granularity I will revisit it. This means that a bigger refactor might be required inside the `lte_link_control` library in a way to separate the different functionality domains across multiple files. Just adding more `ifdefs` across the library will only make it harder to manage in my opinion. I'll mull it over.

I've also had this modularization in my mind especially during the spring time when doing some changes to the library. I wasn't too far from starting it. Time and also the test coverage for the library was a little low for my liking so I didn't do it back then. I was kind of hoping to raise the test coverage and modularize the library later but it could well be done immediately.

But we should consider whether modularization is not with a different PR than Kconfig options which are set to 'n' by default to isolate the breaking changes into a smaller PR.